src/ddmd/s2ir.d

```diff
@@ -557,77 +557,6 @@ extern (C++) class S2irVisitor : Visitor
             return;
         }
 
-        if (s.condition.type.isString())
```

Comment:
Shouldn't this be replaced with an assert? This function no longer handles string conditions, and it would be a compiler error if it were given one.